nanoFramework.Tools.DebugLibrary.Shared/PortSerial/PortSerial.cs

```diff
@@ -153,10 +123,20 @@ public bool IsDeviceConnected
         public void CloseDevice()
         {
             if (NanoDevice.DeviceBase != null)
-            {
-                ((SerialPort)NanoDevice.DeviceBase).Close();
-                ((SerialPort)NanoDevice.DeviceBase).Dispose();
-            }
+            {
+                try
+                {
+                    if (((SerialPort)NanoDevice.DeviceBase).IsOpen)
+                    {
+                        ((SerialPort)NanoDevice.DeviceBase).Close();
+                    }
+                    ((SerialPort)NanoDevice.DeviceBase).Dispose();
+                }
+                catch (Exception ex)
+                {
+                    Debug.WriteLine($">>>> CloseDevice ERROR from {InstanceId}: {ex.Message}");

```

Comment:
Rather than the Debug, possible to use the `OnLogMessageAvailable`?

Reply:
Unfortunately, the OnLogMessageAvailable does not always work! But that's another problem.